server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -1993,4 +1992,23 @@ ON CONFLICT (real_id, \"date\", year)
             "2030| 99998.0"
         );
     }
+
+    @Test
+    public void test_non_deterministic_sub_column_unused_in_targets() {
+        execute("""
+            CREATE TABLE tbl (
+                a int PRIMARY KEY,
+                o1 object as (
+                    sub int as round((random() + 1) * 100)
+                )
+            )
```

Comment:
double checking: this creates at least one replica? (test is annotated with 2 datanodes)

Reply:
yes, by default number_of_replicas is 0-1 (0 on a single node and on 2 nodes it's 1 replica). And without the fix I see ArrayOutOfBound in the addGeneratedValues which means code sensing requests to replicas kicks in.